connectivity/check/deployment.go

```diff
@@ -451,9 +455,83 @@ func (ct *ConnectivityTest) validateDeployment(ctx context.Context) error {
 		}
 	}
 
+	for _, cp := range ct.ciliumPods {
+		err := ct.waitForIPCache(ctx, cp)
```

Comment:
In the worst case, we'll end up waiting for # of cilium pods * ipcache timeout. This logic can be parallelized, no?

Reply:
Maybe, but I don't think complicating the code here is worth it. The error check here will return from the loop as soon as the first Cilium pod times out. I'll look into setting a deadline instead a timeout so that the timeouts are not cumulative.

Reply:
Ended up setting the timeout once before looping over the Cilium pods and changing`waitForIPCache()` try at least once before timing out. This way we check each Cilium Pod's ipcache at least once, so the timeout only limits retries.